integ-test/src/test/java/com/amazon/opendistroforelasticsearch/sql/ppl/QueryAnalysisIT.java

```diff
@@ -0,0 +1,131 @@
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.ppl;
+
+import com.amazon.opendistroforelasticsearch.sql.exception.SemanticCheckException;
+import java.io.IOException;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.ResponseException;
+import org.junit.Ignore;
+import org.junit.Test;
+import static com.amazon.opendistroforelasticsearch.sql.esintgtest.TestsConstants.TEST_INDEX_ACCOUNT;
+import static org.elasticsearch.rest.RestStatus.OK;
+
+public class QueryAnalysisIT extends PPLIntegTestCase {
+
+    @Override
+    public void init() throws IOException {
+        loadIndex(Index.ACCOUNT);
+    }
+
+    /**
+     * Valid commands should pass semantic check
+     */
+    @Test
+    public void searchCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void whereCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | where age=20", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void fieldsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | fields firstname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Ignore("Can't resolve target field yet")
+    @Test
+    public void renameCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | rename firstname as first", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void statsCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | stats avg(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void dedupCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | dedup firstname, lastname", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void sortCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | sort age", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+    @Test
+    public void evalCommandShouldPassSemanticCheck() {
+        String query = String.format("search source=%s | eval age=abs(age)", TEST_INDEX_ACCOUNT);
+        queryShouldPassSemanticCheck(query);
+    }
+
+
+    /**
+     * Commands that fail semantic analysis should throw {@link SemanticCheckException}
+     */
+    @Test
+    public void unsupportedAggregationShouldFailSemanticCheck() {
+        String query = String.format("search source=%s | stats count(age)", TEST_INDEX_ACCOUNT);
```

Comment:
Is count being added in https://github.com/opendistro-for-elasticsearch/sql/pull/493?

Reply:
Thanks for the reminder. Changed the case to `range`, is that making sense?

Reply:
Sure. Thanks!